registry/storage/driver/s3/s3.go

```diff
@@ -367,6 +376,8 @@ func (d *driver) WriteStream(path string, offset int64, reader io.Reader) (total
 		}
 
 		go func(bytesRead int, from int64, buf []byte) {
+			defer d.putbuf(buf) // this buffer gets dropped after this call
```

Comment:
We're going to end up putting the same ```buf``` twice. The ```d.putbuf(buf)``` further up will be bound to the same instance of ```buf```